include/linux/kfence.h

```diff
@@ -68,9 +68,7 @@ static inline bool kfence_discard_slab(struct kmem_cache *s, struct page *page)
 static inline bool kfence_handle_page_fault(unsigned long addr) { return false; }
 static inline bool is_kfence_addr(void *addr) { return false; }
 static __always_inline void *kfence_alloc(struct kmem_cache *s, size_t size, gfp_t flags) { return NULL; }
-static inline bool kfence_free(struct kmem_cache *s, struct page *page,
-			       void *head, void *tail, int cnt,
-			       unsigned long addr) { return false; }
+static inline bool kfence_free(void *addr) { return false; }
```

Comment:
Nice catch!
What do you think about inline vs. __always_inline in our code?
I don't see much difference in practice (though some LWN sources claimed not all compilers respect `inline`), but we probably need to pick one for consistency.

Reply:
Right, we need to also be consistent with other kernel code.

The rule I've been following with KCSAN at least (after some iterations) was:
- Use __always_inline if it's a bug if it is not inlined (i.e. __always_inline is the "better macro"). This can be a performance bug but also things like uaccess warnings that might appear if not inlined.
- Use inline by default. The compiler might know better.
- For empty stubs, use 'inline' because any compiler that doesn't throw away empty inline functions has a bug.

For the stubs we should use 'inline' which is also consistent with other stubs across the kernel. I'll add a commit to fix.